src/Orchard/Tasks/Locking/Services/DistributedLock.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Threading;
+
+namespace Orchard.Tasks.Locking.Services {
+
+    /// <summary>
+    /// Represents a distributed lock. />
+    /// </summary>
+    public class DistributedLock : IDisposable {
+        private IDistributedLockService _service;
+        private int _isDisposed;
+
+        private DistributedLock() {
+        }
+
+        public string Id { get; private set; }
```

Comment:
The idea behind this is that although we are currently storing an ID value of the `DistributedLockRecord` record (which happens to be an `Int32`), the `DistributedLock` itself is agnostic of the actual storage device. Should that ever change, we would not be tied to an Int32 (because maybe other storage implementations want to use GUIDs for example. Anyway, that is all theoretic. In practice I doubt it will ever change, but I still think it's good that the ID property of the `DistributedLock` is agnostic of `DistributedLockRecord`. That being said, I don;t really care, so if you feel strongly about using an `Int32` I would be happy to change it